packages/discovery-react-components/src/components/StructuredQuery/components/RuleGroup/RuleGroup.tsx

```diff
@@ -0,0 +1,85 @@
+import React, { FC, Dispatch, SetStateAction } from 'react';
+import { Messages } from 'components/StructuredQuery/messages';
+import { RuleGroupDropdown } from '../RuleGroupDropdown/RuleGroupDropdown';
+import { RuleRow } from '../RuleRow/RuleRow';
+import { AddRuleRowButton } from '../AddRuleRowButton/AddRuleRowButton';
+import {
+  StructuredQuerySelection,
+  Row
+} from 'components/StructuredQuery/utils/structuredQueryInterfaces';
+import { MAX_NUM_SIBLING_RULE_ROWS } from 'components/StructuredQuery/constants';
+import {
+  structuredQueryRuleGroupClass,
+  structuredQueryNestedRuleGroupClass,
+  structuredQueryRulesButtonsClass
+} from 'components/StructuredQuery/cssClasses';
+
+export interface RuleGroupProps {
+  /**
+   * override default messages for the component by specifying custom and/or internationalized text strings
+   */
+  messages: Messages;
+  /**
+   * id of the group for the rule row to render, or 'top-level' if the top-level rule group
+   */
+  groupId: number | 'top-level';
+  /**
+   * state that represents the current rules and selections for the structured query
+   */
+  structuredQuerySelection: StructuredQuerySelection;
+  /**
+   * used to set the structuredQuerySelection state
+   */
+  setStructuredQuerySelection: Dispatch<SetStateAction<StructuredQuerySelection>>;
+}
+
+export const RuleGroup: FC<RuleGroupProps> = ({
+  messages,
+  groupId,
+  structuredQuerySelection,
+  setStructuredQuerySelection
+}) => {
+  const isTopLevelGroup = groupId === 'top-level';
+  let rows: Row[] = structuredQuerySelection.rows;
+  if (!isTopLevelGroup) {
+    structuredQuerySelection.groups.map(group => {
+      if (group.id === groupId) {
+        rows = group.rows;
+      }
+    });
+  }
+  const showAddRuleRowButton = rows.length < MAX_NUM_SIBLING_RULE_ROWS && !isTopLevelGroup;
```

Comment:
do we not want to allow a rule addition at the top level?

Reply:
I'm checking for this here since we already include the `Add rule` button for the top-level in `StructuredQuery` to make its `Add rule` button appear all the way at the bottom. I could add a comment if you think that would help clarify in future?

Reply:
i think its fine. i'm guessing that if a rule group only has a single rule we don't wrap parentheses around the query expression that we end up generating